vite-project-kabinet/src/Content/LoadedList.jsx

```diff
@@ -1,22 +1,78 @@
 import "./paginated.style.css"
 import { CardPreview } from "./CardPreview.jsx"
 import { useState } from "react";
+import { useEffect } from "react";
 
-export const LoadedList = ({items, itemsPerPage}) => {
+export const LoadedList = ({itemsPerPage}) => {
+    const [items, setItems] = useState([])
+    const [totalRecords, setTotalRecords] = useState(0)
     const [visibleCount, setVisibleCount] = useState(itemsPerPage)
 
+    const getPostsWithCategoryName = async (perPage) => {
+
+        const reqPosts = await fetch(`https://casopiskabinet.sk/wp-json/wp/v2/posts?per_page=${perPage}&page=1`)
+        const totalRecords = reqPosts.headers.get("X-WP-Total")
+        setTotalRecords(totalRecords)
+        const posts = await reqPosts.json()
+    
+        const reqCategories = await fetch(`https://www.casopiskabinet.sk//wp-json/wp/v2/categories?_fields=id,name`)
+        const categories = await reqCategories.json()
+    
+        //chcem získať pole objektov, kde bude každý objekt obsahovať aj konkrétny názov kategórie
+        
+        const updatedPosts = posts.map((post) => {
+            const postCategsArray = post.categories
+            const newPostCategsArray = postCategsArray.map((catID) => {
+                const categObject = categories.find((category) => category.id === catID )
+                return categObject ? categObject.name : "category not found"
+            })
+    
+            // Aby program správne fungoval, každý článok musí obsahovať vždy práve jednu kategóriu.
+            // Kabinet má aktuálne pár článkov takých, kde sú pridelené 2 kategórie
+    
+            // todo: Dá sa nastaviť WP tak, aby uživateľ mohol zakliknúť vždy len jednu kategóriu ?
+            // Nedá. Muselo by se vytvořit nové pole, které by fungovalo stejně jen s tímhle omezením. Znamenalo by to, že by se musely upravit hodnoty pro všechny články.
+
+            // todo: pridať žánre vo WP
+            // todo: nastaviť WP tak, aby šlo pridávať fotky
+            // todo: nastaviť WP tak, aby mal autor svoju vlastnú kolonku
+
+            const stripHtmlTags = (html) => {
```

Comment:
Tohle je docela komplikované. HTML značky můžeš odstranit i pomocí regulárních výrazů, viz https://www.30secondsofcode.org/js/s/strip-html-tags/#:~:text=All%20you%20really%20need%20is,and%20finally%20the%20closing%20bracket.

A teď tu funkci definuješ znovu pro každý záznam v `posts`. Tahle funkce by měla být definována mimo komponentu, aby se vytvořila jen jednou. Většinou se to dává do vlastních _utils_ souborů, kde jsou různí podobní pomocníci.